### PR TITLE
framework/include/st_things/st_things_types.h : Fix the doxygen error

### PR DESCRIPTION
**Pull request by KimDongEon**
Fixed the doxygen data structure name recognition error

_st_things_get_request_message -> st_things_get_request_message
_st_things_representation -> st_things_representation
_st_things_set_request_message -> st_things_set_request_message

### CODE REVIEW

File: framework/include/st_things/st_things_types.h [hunk 1]
```diff
@@ -297,12 +297,13 @@ typedef struct _st_things_representation {
 	 bool(*set_object_array_value) (struct _st_things_representation *rep, const char *key, const struct _st_things_representation **array, size_t length);
 
 } st_things_representation_s;
+typedef _st_things_representation_s st_things_representation_s;
```
Comment (jeongchanKim) [inline framework/include/st_things/st_things_types.h]: please follow the below
struct  name : XXX_s
typedef name : XXX_t

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/include/st_things/st_things_types.h (hunk 1 above)
Comment (KimDongEon) [reply framework/include/st_things/st_things_types.h]: Change. Please review.

**Comment by seinfra** [comment]
Code Rule Check OK.

File: framework/include/st_things/st_things_types.h [hunk 2]
```diff
@@ -297,12 +297,13 @@ typedef struct _st_things_representation {
 	 bool(*set_object_array_value) (struct _st_things_representation *rep, const char *key, const struct _st_things_representation **array, size_t length);
 
 } st_things_representation_s;
```
Comment (sunghan-chang) [inline framework/include/st_things/st_things_types.h]: You shoud remove xxx_s from here.

File: framework/include/st_things/st_things_types.h [hunk 3]
```diff
@@ -327,12 +328,13 @@ typedef struct _st_things_get_request_message {
 	 bool(*has_property_key) (struct _st_things_get_request_message *req_msg, const char *key);
 
 } st_things_get_request_message_s;
```
Comment (sunghan-chang) [inline framework/include/st_things/st_things_types.h]: same as previous

File: framework/include/st_things/st_things_types.h [hunk 4]
```diff
@@ -348,6 +350,7 @@ typedef struct _st_things_set_request_message {
 	 bool(*get_query_value) (struct _st_things_set_request_message *req_msg, const char *key, char **value);
 
 } st_things_set_request_message_s;
```
Comment (sunghan-chang) [inline framework/include/st_things/st_things_types.h]: same as previous

**Comment by sunghan-chang** [comment]
@KimDongEon Please revert your latest commit for update your repo. And write commit title and description details.

**Pull request closed without merging**